patches/api/0462-Adding-PredicateChoice-to-Paper-API.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: derverdox <mail.ysp@web.de>
+Date: Mon, 4 Dec 2023 01:54:38 +0100
+Subject: [PATCH] Adding PredicateChoice to Paper API
+
+
+diff --git a/src/main/java/org/bukkit/inventory/RecipeChoice.java b/src/main/java/org/bukkit/inventory/RecipeChoice.java
+index 523818cbb0d6c90481ec97123e7fe0e2ff4eea14..6b9bbead57166d7622cf31b96c252d56fafc05fb 100644
+--- a/src/main/java/org/bukkit/inventory/RecipeChoice.java
++++ b/src/main/java/org/bukkit/inventory/RecipeChoice.java
+@@ -233,4 +233,64 @@ public interface RecipeChoice extends Predicate<ItemStack>, Cloneable {
+             return "ExactChoice{" + "choices=" + choices + '}';
+         }
+     }
++
++    /**
++     * Represents a choice that matches when the item predicate is fulfilled.
++     */
++    public static class PredicateChoice implements RecipeChoice {
```

Comment:
This should be an interface that is backed by a package-private record implementation with static methods to create instances of it.